src/components/update-service/update-service.controller.ts

```diff
@@ -299,6 +253,37 @@ export class UpdateServiceController implements angular.IController {
     }
   }
 
+
+  private updateServicePlanName(instance: object) {
+    if (_.get(instance, 'spec.clusterServicePlanExternalName') !== _.get(this.ctrl.selectedPlan, 'spec.externalName')) {
+      _.unset(instance, 'spec.clusterServicePlanRef');
+      _.set(instance, 'spec.clusterServicePlanExternalName', _.get(this.ctrl.selectedPlan, 'spec.externalName'));
+    }
+  };
+
+  private updateSpecParameters(instance: object, updateParameters: object, originalParameters: object) {
+    if (!angular.equals(updateParameters, originalParameters)) {
+      _.set(instance, 'spec.parameters', updateParameters);
+    }
+  };
+
+  private updateSecretParameters(instance: object, newParameters: object) {
+    let secretParams: any = {};
```

Comment:
A lot of these `any` declarations are not needed. It will infer the type based on assignment.